src/components/ViewSource/ViewSource.js

```diff
@@ -29,6 +29,8 @@ const ViewSource = ({
   const [isShownPrintPopup, setIsShownPrintPopup] = useState(false);
   const openPrintPopup = () => setIsShownPrintPopup(true);
   const closePrintPopup = () => setIsShownPrintPopup(false);
+  const stripes = useStripes();
+  const isPrintAvailable = stripes.hasPerm('ui-quick-marc.quick-marc-editor.view') || stripes.hasPerm('ui-quick-marc.quick-marc-holdings-editor.all');
```

Comment:
A user shouldn't be able to print the `holdings` record without `ui-quick-marc.quick-marc-holdings-editor.all` permission.

Reply:
Nice catch.